### PR TITLE
Automate doc copy

### PR DESCRIPTION
**Pull request by kne42**
# Description
Automates the copying of doc sources from `napari/napari` to `napari/napari.github.io`. Also updates the table of contents in `napari.github.io`'s `_toc.yml`.

Note: the formatting for the autogenerated ToC is super ugly, but still preserves ordering. We could make it prettier using [`ruamel.yaml`](https://yaml.readthedocs.io/en/latest/) but I feel like it's not really worth the effort.

The file contains configuration variables for which files to include and which ones to ignore (including a separate list of files to ignore when updating the ToC).

### CODE REVIEW

File: docs/copy-docs.py [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+    """Exclude files in the ignore list and duplicated files."""
+    for ignore in IGNORE:
+        if fnmatch(path, osp.join(SRC, ignore)):  # in ignore list
+            return True
+    else:
+        if osp.isdir(path) or osp.splitext(path)[1] != '.md':
+            return False
+        with open(path) as f:
+            firstline = f.readline()
+        return firstline.startswith('```{include}')  # duplicate file
```
Comment (codemonkey800) [inline docs/copy-docs.py:66]: i'm curious, why does **```{include}`** indicate a duplicate file? 🤔

**Review by codemonkey800 — COMMENTED** [review]
Looking good to me. I'll let someone else with more context approve 😄

**Review by sofroniewn — APPROVED** [review]
Changes look good to me!! Maybe @tlambert03 can give an additional look and then we can merge

File: docs/copy-docs.py (hunk 1 above)
Comment (kne42) [reply docs/copy-docs.py:66]: basically to preserve the structure for the docs at https://napari.org/docs, I have files which are nothing more than
````markdown
```{include} filepath
```
````

that effectively point to where the files are actually located for the main https://napari.org site

so for example we used to have a file at `developer/roadmap_0_3` but now we want it at `roadmaps/0_3` so to preserve the old structure, `roadmaps/0_3` is not in the table of contents here, but it is in `napari.github.io`

**Review by tlambert03 — APPROVED** [review]
looks good to me!  only very minor comment is that `pathlib.Path` could come in handy here (e.g. `osp.join(osp.dirname(osp.dirname(osp.dirname(__file__))), 'napari.github.io')` turns to `Path(__file__).parent.parent.parent / 'napari.github.io'` ..

It might also be nice to have a general overview comment somewhere that just summarizes how this whole thing works.  like "Whenever we push to master, we automatically commit the contest of ____ to `napari.github.io` ... that commit will then trigger a workflow there, etc...".  
(I'm also not sure if there's some explanation somewhere already of how versions/stable/dev works, but that would be nice too.)

thanks @kne42!

**Comment by justinelarsen** [comment]
Looks great @kne42 , nice work!

I like what @tlambert03 suggested with the explanation of how it all works too. Would be great to include that.

**Comment by kne42** [comment]
I've added documentation on how everything is copied/built. You can preview it [here](https://github.com/kne42/napari/blob/automate-doc-copy/docs/ORGANIZATION.md).

**Comment by tlambert03** [comment]
perfect, thanks!

**Comment by sofroniewn** [comment]
Amazing, thanks so much @kne42 - the additional doc is so helpful - will merge now!!

**Pull request merged**